drivers/can/can_nxp_s32_canxl.c

```diff
@@ -0,0 +1,1097 @@
+		.prop_seg = 0x08,
+		.phase_seg1 = 0x08,
+		.phase_seg2 = 0x08,
+		.prescaler = 0x100
+	}
+#endif
+};
+
+#define CAN_NXP_S32_NODE(n)			DT_NODELABEL(can##n)
+
+#define CAN_NXP_S32_IRQ_HANDLER(n, irq_name)	DT_CAT5(CANXL, n, _, irq_name, Handler)
+
+#define _CAN_NXP_S32_IRQ_CONFIG(node_id, prop, idx, n)					\
+	do {										\
+		extern void (CAN_NXP_S32_IRQ_HANDLER(n,					\
+				DT_STRING_TOKEN_BY_IDX(node_id, prop, idx)))(void);	\
+		IRQ_CONNECT(DT_IRQ_BY_IDX(node_id, idx, irq),				\
+				DT_IRQ_BY_IDX(node_id, idx, priority),			\
+				CAN_NXP_S32_IRQ_HANDLER(n,				\
+					DT_STRING_TOKEN_BY_IDX(node_id, prop, idx)),	\
+				NULL,							\
+				DT_IRQ_BY_IDX(node_id, idx, flags));			\
+		irq_enable(DT_IRQ_BY_IDX(node_id, idx, irq));				\
+	} while (false);
+
+#define CAN_NXP_S32_IRQ_CONFIG(n)							\
+	static void can_irq_config_##n(void)						\
+	{										\
+		DT_FOREACH_PROP_ELEM_VARGS(CAN_NXP_S32_NODE(n), interrupt_names,	\
+							_CAN_NXP_S32_IRQ_CONFIG, n);	\
+	}
+
+#define CAN_NXP_S32_ERR_CALLBACK(n)							\
+	void nxp_s32_can_##n##_err_callback(uint8 instance, Canexcel_Ip_EventType eventType,\
+		uint32 u32SysStatus, const Canexcel_Ip_StateType *canexcelState)	\
+	{										\
+		const struct device *dev = DEVICE_DT_GET(CAN_NXP_S32_NODE(n));		\
+		can_nxp_s32_err_callback(dev, eventType, u32SysStatus, canexcelState);	\
+	}
+
+#define CAN_NXP_S32_CTRL_CALLBACK(n)							\
+	void nxp_s32_can_##n##_ctrl_callback(uint8 instance, Canexcel_Ip_EventType eventType,\
+			uint32 buffIdx, const Canexcel_Ip_StateType *canexcelState)	\
+	{										\
+		const struct device *dev = DEVICE_DT_GET(CAN_NXP_S32_NODE(n));		\
+		can_nxp_s32_ctrl_callback(dev, eventType, buffIdx, canexcelState);	\
+	}
+
+#if defined(CONFIG_CAN_FD_MODE)
+#define CAN_NXP_S32_TIMING_DATA_CONFIG(n)						\
+		.bitrate_data = DT_PROP(CAN_NXP_S32_NODE(n), bus_speed_data),		\
+		.sjw_data = DT_PROP(CAN_NXP_S32_NODE(n), sjw_data),			\
+		.prop_seg_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), prop_seg_data, 0),	\
+		.phase_seg1_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), phase_seg1_data, 0),	\
+		.phase_seg2_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), phase_seg2_data, 0),	\
+		.sample_point_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), sample_point_data, 0),
+#define CAN_NXP_S32_FD_MODE	1
+#define CAN_NXP_S32_BRS		1
+#else
+#define CAN_NXP_S32_TIMING_DATA_CONFIG(n)
+#define CAN_NXP_S32_FD_MODE	0
+#define CAN_NXP_S32_BRS		0
+#endif
+
+#ifdef CONFIG_CAN_AUTO_BUS_OFF_RECOVERY
+#define CAN_NXP_S32_CTRL_OPTIONS CANXL_IP_BUSOFF_RECOVERY_U32
+#else
+#define CAN_NXP_S32_CTRL_OPTIONS 0
+#endif
+
+#define CAN_NXP_S32_INIT_DEVICE(n)							\
+	CAN_NXP_S32_CTRL_CALLBACK(n)							\
+	CAN_NXP_S32_ERR_CALLBACK(n)							\
+	CAN_NXP_S32_IRQ_CONFIG(n)							\
+	PINCTRL_DT_DEFINE(CAN_NXP_S32_NODE(n));						\
+	Canexcel_Ip_ConfigType can_nxp_s32_default_config##n = {			\
+		.rx_mbdesc = (uint8)CONFIG_CAN_NXP_S32_MAX_RX,				\
+		.tx_mbdesc = (uint8)CONFIG_CAN_NXP_S32_MAX_TX,				\
+		.CanxlMode = CANEXCEL_LISTEN_ONLY_MODE,					\
```

Comment:
done https://github.com/zephyrproject-rtos/zephyr/pull/52450#discussion_r1178317471
Please point me to where this is done?